### DISCUSSION TITLE
Project members update

### DISCUSSION THREAD

**Discussion by Jakuje** [OpenSC/OpenSC#2966] [General]
I started this discussion in email last week that the current list of maintainers and core members is no longer up-to-date, making some of the arrangements complicated (for example assigning correct people to the confidential security reports on github) but for the sake of transparency, I am starting an public discussion about this topic, but would like to hear opinions mostly from @OpenSC/core, but also from anyone else who would like to point out some valid reasons for not doing these changes or if I missed somebody.

From the current list of @OpenSC/opensc-maintainers : 
 * I see that @szikora is not much active over last ~10 years. Would it make sense to remove him from the maintainers (or do you use your role in some other way that I do not see?)
 * On the other hand, I see @popovec quite active in all contributions, reviews and discussions over last years so I would be proposing to add him to this group (indeed only if you are interested!).
 * My last proposal would be to add @xhanulik. She extended the fuzzing of the OpenSC over the last years, but touched several other areas including huge improvement of the support for IDPrime cards. I would like her to take over some work I do these days
 * There are also @metsma and @hhonkanen having fewer contributions, but still quite consistent over the last years, but mostly in their area of expertise.

Mostly based on the following report: https://github.com/OpenSC/OpenSC/graphs/contributors

**Comment by frankmorgner** [comment]
I agree to the suggestions, given there are no objections from the people mentioned.

**Comment by popovec** [comment]
As long as it means for me to continue what I have been doing so far, ie. track issues, contribute to fixing them and add new features, so of course I agree. At the same time, I also agree with the other proposed people.

**Reply by Jakuje** [reply]
I do not think we have any formal responsibilities defined so what you describe basically matches what we do. But in addition, we do releases and we have access to embargoed issues and merge requests where another pair of eyes is always helpful. We can also add you to coverity scan (I see you are there already) and oss-fuzz reports if you would be interested. That are also lists somewhere else that needs to be re-checked:

https://github.com/google/oss-fuzz/blob/master/projects/opensc/project.yaml

**Reply by popovec** [reply]
Of course, I am also interested in reports from oss-fuzz. And I fully understand that in the case of security problems, it is necessary not to publish anything that would allow the security problem to be exploited until the given problem is solved.

**Reply by Jakuje** [reply]
Adding Peter to the oss-fuzz here: https://github.com/google/oss-fuzz/pull/11392

**Comment by frankmorgner** [comment]
With more members on the board, I hope, we can accept/review pull requests more quickly. Better branch protection rules should then also be possible.

**Comment by LudovicRousseau** [comment]
No objections on my side. It is often a good idea to embark active people.
I was not very active on OpenSC in the last years.

I also propose to add @Jakuje to @OpenSC/core just in case the other core members "disappear".
I don't know why we have @OpenSC/core and @OpenSC/maintainers as they both have the "Admin" role.

**Reply by Jakuje** [reply]
Thank you! I agree that having me in the core would be a good addition :) I requested to join the core. Can you take care of approving/adding me?

I see the only difference between Maintainers and Core, that the core has https://github.com/OpenSC/OpenSC.tokend repository admin access and Martin instead of me in members so my proposal would be to drop the Maintainers team if we will not be able to figure out why it should exist.

**Reply by LudovicRousseau** [reply]
@Jakuje you are now a member of Core.

**Reply by Jakuje** [reply]
Thank you. Got back to that just today.

Not sure how the permissions work but I see myself in https://github.com/orgs/OpenSC/teams/core?query=role%3Amember but only with a role member, while everyone else as a maintainer.

I would like to invite the others mentioned in this thread, but I do not see that option anywhere, likely because of different membership.

**Reply by frankmorgner** [reply]
please check again, you should now be admin

**Reply by Jakuje** [reply]
Thank you. Now I see there much more options.

**Reply by frankmorgner** [reply]
Core/Maintainers seem to serve the same purpose. I've checked "Settings > Collaborators and Teams" for every repository in the OpenSC organizaition and Core/Maintainers were present as Admin everywhere except in tokend (Maintainers was missing). So I think the group Maintainers is superfluous and can be removed. This then also avoids the possible confusion with XXX maintainers, who all have "write" access instead of admin rights.

**Reply by Jakuje** [reply]
Thanks. I invited @xhanulik and @popovec for now to OpenSC maintainers.

Next step would be probably removing the Maintainers group if there will be no strong voices against.

I would like also to remove the `JenkinsForOpenSC` and configure the organization to require everyone to have 2FA enabled (I see it is now everyone except for this user -- is it used for anything now? Do somebody have access to that? I see .fr domain in the bio so my guess would be @LudovicRousseau).

**Reply by Jakuje** [reply]
Just enabled the requirement for 2FA for everyone. I think it should not affect anyone, but just noting it here as it was discussed.

**Comment by metsma** [comment]
No objections on my side.

**Reply by Jakuje** [reply]
Sent invite today (sorry for a delay)

**Comment by hhonkanen** [comment]
No objections on my side.

**Reply by Jakuje** [reply]
@metsma @hhonkanen in case you are interested in oss-fuzz and coverity scan reports on master, please, send me your mails (either here or to my mail). The only limitation is that oss-fuzz requires gmail account to access all the functionality and it will be in the public oss-fuzz repository.

For access to the opensc maintainers, either request the access on https://github.com/orgs/OpenSC/teams/opensc-maintainers or wait for the invitation,

**Reply by Jakuje** [reply]
Sent invite now. Sorry for a delay.

**Comment by xhanulik** [comment]
No objections, either.

**Comment by LudovicRousseau** [comment]
> I would like also to remove the `JenkinsForOpenSC` and configure the organization to require everyone to have 2FA enabled (I see it is now everyone except for this user -- is it used for anything now? Do somebody have access to that? I see .fr domain in the bio so my guess would be @LudovicRousseau).

`JenkinsForOpenSC` was used by @viktorTarasov for its opensc.fr/jenkins service. See https://github.com/OpenSC/OpenSC/issues/595#issuecomment-188230246
I think we can remove this account now.

**Reply by Jakuje** [reply]
This was removed too.

**Comment by Jakuje** [comment]
I think all of the items in this discussion were resolved so closing now. Thank you.
